distributed/scheduler.py

```diff
@@ -8570,9 +8601,17 @@ def _materialize_graph(
             deps.remove(k)
         dependencies[k] = deps
 
-    # Remove aliases
+    # FIXME: Remove this section. This condition is generated by
+    # unpack_remotedata when using persisted collections. There should be a
+    # better way to achieve that tasks are not self referencing themselves
+
+    # Remove aliases.
     for k in list(dsk):
-        if dsk[k] is k:
+        # Most keys are tuples so the `is` comparison does not work
+        # However, we cannot simply use `==` since dsk[k] may contain actual
+        # payload data, e.g. a pandas DataFrame where equals is not necessarily
+        # properly defined with any object
+        if dsk[k] is k or tokenize(dsk[k]) == tokenize(k):
             del dsk[k]
```

Comment:
This is somewhat a regression from the stringification change. Essentially, there are graphs where a key can map to itself and this section is supposed to clean this up.
This didn't work since the stringification change (and maybe didn't even before).

However, now, this is important since we'd otherwise trigger the run_spec check I introduced

cc @crusaderky this is what I was talking about earlier

Reply:
I don't think it worked before?
To my understanding, the only two cases where `a is b` with two strings is that (a) the string is internalized, which _should_ not be the case for strings of non-trivial length or (b) the two strings were the same object on the client and were serialized with pickle (msgpack would deep-copy them).